dts/arm/atmel/sam4e16c.dtsi

```diff
@@ -7,11 +7,12 @@
 #include <mem.h>
 #include <atmel/sam4e.dtsi>
 
-/ {
-	flash0: flash@400000 {
-		reg = <0x00400000 DT_SIZE_K(1024)>;
-	};
+&flash0 {
+	reg = <0x00400000 DT_SIZE_K(1024)>;
+};
```

Comment:
I did some tests with full path and works fine because of MCUboot. In general, we I saw constructions like:
```
/ {
	soc {
		flash-controller@400e0a00 {
			compatible = "atmel,sam-flash-controller";
			label = "FLASH_CTRL";
			reg = <0x400e0a00 0x200>;
			peripheral-id = <6>;

			#address-cells = <1>;
			#size-cells = <1>;

			flash0: flash@400000 {
				compatible = "soc-nv-flash";
				label = "FLASH_0";

				write-block-size = <16>;
				erase-block-size = <8192>;
			};
		};
```
It seems more clear when define the full path:
```
/ {
	soc {
		flash-controller@400e0a00 {
			flash0: flash@400000 {
				reg = <0x00400000 DT_SIZE_K(1024)>;
			};
		};
	};
};
```

Reply:
Yeah, I was being a bit lazy and using the node label reference form so that way the location/path didn't have to be explicit.

Reply:
Changed style back to show full SoC path.  Also moved the sram nodes while we're at it.